app/models/concerns/hyku_addons/user_behavior.rb

```diff
@@ -21,10 +21,14 @@ def current_account
       Site.account
     end
 
+    def display_profile_visibility
+      user.display_profile ? "open" : "closed"
+    end
+
     def toggle_display_profile
       return unless display_profile_changed?
       yield
-      HykuAddons::ToggleDisplayProfileJob.perform_later(email, display_profile)
+      HykuAddons::ToggleDisplayProfileJob.perform_later(email, display_profile_visibility)
```

Comment:
Do you need to change this? `display_profile_visibility` is the string used in the JSON, rather than the users boolean preference.

Reply:
Yeah because the Job changes the JSON string in all the works, otherwise you are doing the conditional again

Reply:
I see, so the job updates the JSON to whatever value is passed in here. Cool :)